bankline_parser/data_services/fields.py

```diff
@@ -0,0 +1,96 @@
+from datetime import datetime
+
+from .exceptions import ParseError
+
+
+class DataField(object):
+
+    def __init__(self, start, end, justification='r', fill_char=' ', pad_char=' '):
+        self.start = start
+        self.end = end
+        self.justification = justification
+        self.fill_char = fill_char
+        self.pad_char = pad_char
+
+    def _strip_padding(self, field_content):
+        if self.justification == 'r':
+            for i in range(len(field_content)):
+                if field_content[i] == self.pad_char:
+                    continue
+                else:
+                    return field_content[i:]
+        else:
+            for i in reversed(range(len(field_content))):
+                if field_content[i] == self.pad_char:
+                    continue
+                else:
+                    return field_content[:i+1]
+
+    def parse(self, row_content):
+        field_content = row_content[self.start:self.end]
+        if field_content == self.fill_char*(self.end-self.start):
+            return None
+        return self._strip_padding(field_content)
+
+
+class TextField(DataField):
+    pass
+
+
+class DateField(DataField):
+
+    def parse(self, row_content):
+        field_content = row_content[self.start:self.end]
+        try:
+            return datetime.strptime(field_content, ' %y%j')
+        except ValueError as e:
+            raise ParseError(e)
+
+
+class NumericField(DataField):
+
+    def parse(self, row_content):
+        field_content = super(NumericField, self).parse(row_content)
+        if field_content:
+            try:
+                return int(field_content)
+            except (TypeError, ValueError) as e:
+                raise ParseError(e)
+        return 0
```

Comment:
No. It ended up that way because of one field, the 'count of balance records', which if not used is blank filled, which effectively means that there are 0 balance records. Was just easier for validation to return 0.

I guess maybe you're right, though.